### PR TITLE
menuentryswapper: Hop to option for the menu swapper plugin.

### LINKED ISSUE #1809: MES: Swap "Kick User" with "Hop-to" for CC
**Is your feature request related to a problem? Please describe.**
When you have kick power in a Clan Chat your default hover option is "kick user", what I'd like is the ability to change this to "hop-to" so I can't accidentally kick people when hopping world for raids etc VIA the clan interace

**Describe the solution you'd like**
A menu entry swap that changes "kick user" with "hop-to" that doesn't affect the friendslist.

I had this done using the old MES style where you could define the target option but I deleted all my files, will make a PR if no one beats me to it.

**Describe alternatives you've considered**
Tried using the custom swaps but it affects the friendslist options too because you can't define a target, just menu options.

**Additional context**
https://i.imgur.com/TrG0LFW.png I cut out the arrow in the screenshot but you get the gist

### PR DESCRIPTION
**Pull request by benjaminlgur**
Allows Hop-to be set as top priority for menu swapper for the friends list, clan chat, or both. The clan chat option closes #1809 . This is my first PR to the open-osrs project and im still learning the api so criticism/advice would be appreciated, so that I can make better and more substantial contributions.

### CODE REVIEW

**Comment by xKylee** [comment]
You haven’t included util.HopToMode

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/util/HopToMode.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2018, Adam <Adam@sigterm.info>
+ * Copyright (c) 2018, Kamiel
+ * Copyright (c) 2019, alanbaumgartner <https://github.com/alanbaumgartner>
+ * Copyright (c) 2019, Kyle <https://github.com/kyleeld>
+ * Copyright (c) 2019, Lucas <https://github.com/lucwousin>
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/util/HopToMode.java]: Thanks but I didn't write this

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 2]
```diff
@@ -1304,6 +1307,30 @@ private void addSwaps()
...
 			menuManager.addPriorityEntry(new GrimyHerbComparableEntry(this.swapGrimyHerbMode, client));
 		}
+
+		switch (this.swapHopToMode)
+		{
+			//.removePriorityEntry needed here to reset Priority
+			case FRIENDSLIST:
+				menuManager.removePriorityEntry("message");
+				menuManager.addPriorityEntry("hop-to").setPriority(1);
+				menuManager.addPriorityEntry("kick user").setPriority(100);
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: I thought the option for kicking was "Kick"? That's what world hopper, hiscores, client scripts, and chat translation seem to use

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/util/HopToMode.java (hunk 1 above)
Comment (benjaminlgur) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/util/HopToMode.java]: Thanks for your comments ill see what I can do.

**Pull request closed without merging**